### PR TITLE
Add approx_count

### PR DESCRIPTION
**Pull request by zengyan1**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #10685 

## What this PR does / why we need it:

### CODE REVIEW

**Review by aunjgr — CHANGES_REQUESTED** [review]
Need some bvt cases

**Review by m-schen — APPROVED** [review]
function lgtm.

**Review by heni02 — APPROVED** [review]

**Review by iamlinjunhong — APPROVED** [review]

File: pkg/sql/plan/query_builder.go [hunk 1]
```diff
@@ -1114,6 +1114,77 @@ func (builder *QueryBuilder) remapSinkScanColRefs(nodeID int32, step int32, sink
...
+							},
+							Expr: &plan.Expr_C{
+								C: &plan.Const{
+									Value: &plan.Const_Sval{
+										Sval: child.ObjRef.SchemaName + "." + child.TableDef.Name,
+									},
+								},
+							},
+						})
+						exprs[0].Typ.Width = int32(len(exprs[0].Expr.(*plan.Expr_C).C.Value.(*plan.Const_Sval).Sval))
```
Comment (aunjgr) [inline pkg/sql/plan/query_builder.go:1155]: you have a more readable way to do this

File: pkg/sql/plan/query_builder.go [hunk 2]
```diff
@@ -1114,6 +1114,77 @@ func (builder *QueryBuilder) remapSinkScanColRefs(nodeID int32, step int32, sink
...
+							},
+						})
+						exprs[1].Typ.Width = int32(len(exprs[1].Expr.(*plan.Expr_C).C.Value.(*plan.Const_Sval).Sval))
+						scanNode := &plan.Node{
+							NodeType: plan.Node_VALUE_SCAN,
+						}
+						childId := builder.appendNode(scanNode, nil)
+						node.Children[0] = builder.buildMetadataScan(nil, nil, exprs, childId)
+						agg.F.Args[0].Expr.(*plan.Expr_Col).Col.RelPos = builder.qry.Nodes[node.Children[0]].BindingTags[0]
+						agg.F.Args[0].Typ = builder.qry.Nodes[node.Children[0]].TableDef.Cols[agg.F.Args[0].Expr.(*plan.Expr_Col).Col.RelPos].Typ
```
Comment (aunjgr) [inline pkg/sql/plan/query_builder.go:1176]: should avoid type assertion as much as possible, for both safety and readability

Comment (aunjgr) [reply pkg/sql/plan/query_builder.go:1176]: and the `RelPos` here is wrong

**Review by aunjgr — CHANGES_REQUESTED** [review]

**Review by aunjgr — APPROVED** [review]

**Pull request merged**